src/main/ruby/core/truffle/cext.rb

```diff
@@ -44,10 +50,33 @@ def self.main(argv = ARGV)
           end
         end
       end
-
+      libraries = resolve_libraries(libraries, search_paths)
       Truffle::CExt.linker(output, libraries, files)
     end
 
+    def self.resolve_libraries(libraries, search_paths)
+      require 'pathname'
+      resolved_libraries = []
+      libraries.each do |lib|
+        if Pathname.new(lib).absolute?
+          resolved_libraries << lib
+        else
+          resolved_libraries << library_in_search_path_or_default(lib, search_paths)
+        end
+      end
+      resolved_libraries
+    end
+
+    def self.library_in_search_path_or_default(lib, search_paths)
+      search_paths.each do |search_path|
+        lib_in_path = File.join(search_paths, lib)
```

Comment:
Typo: `search_path`. Maybe name it `path`/`dir` to be visually more distinctive.